lib/parameters_strapi.py

```diff
@@ -0,0 +1,273 @@
+###
+### Copyright (C) 2024 Intel Corporation
+###
+### SPDX-License-Identifier: BSD-3-Clause
+###
+
+import copy
+import itertools
+
+def gen_avc_cqp_strapi_variants(spec, profiles):
+  for case, params in spec.items():
+    variants = copy.deepcopy(params.get("variants", dict()).get("cqp_strapi", None))
+    if variants is None:
+      keys = ["gop", "slices", "bframes", "qp", "quality", "profile"]
+      product  = list(itertools.product([1], [1], [0], [14, 28], [1, 4, 7], profiles))  # I, single-slice
+      product += list(itertools.product([30], [1], [0], [14, 28], [1, 4, 7], profiles)) # IP, single-slice
+      product += list(itertools.product([30], [4], [2], [14, 28], [1, 4, 7], profiles)) # IPB, multi-slice
+      variants = [dict(zip(keys, vals)) for vals in product]
+
+    for variant in variants:
+      uprofile = variant.get("profile", None)
+      cprofiles = [uprofile] if uprofile else profiles
+
+      # backwards compatibility for ipbmode user config
+      ipbmode = variant.get("ipbmode", None)
+      if ipbmode is not None:
+        assert not variant.get("gop", None)
+        assert not variant.get("slices", None)
+        assert not variant.get("bframes", None)
+        variant.update(
+          slices = 4 if ipbmode == 2 else 1,
+          gop = 30 if ipbmode != 0 else 1,
+          bframes = 2 if ipbmode == 2 else 0)
```

Comment:
This is unnecessary since it is a new variant and therefore has no backwards compat issue.

Reply:
used the consolidated generators for strapi parameters, so there is no this issue now